pkg/planner/core/find_best_task.go

```diff
@@ -1365,6 +1365,13 @@ func (ds *DataSource) FindBestTask(prop *property.PhysicalProperty, planCounter
 	}()
 
 	cntPlan = 0
+	bestTaskStore := kv.TiKV
+	defer func() {
+		if (ds.preferStoreType&h.PreferTiKV > 0 && bestTaskStore != kv.TiKV) ||
+			(ds.preferStoreType&h.PreferTiFlash > 0 && bestTaskStore != kv.TiFlash) {
+			ds.SCtx().GetSessionVars().StmtCtx.AppendWarning(errors.New("the read_from_storage hint might not work"))
```

Comment:
```suggestion
			ds.SCtx().GetSessionVars().StmtCtx.AppendWarning(errors.NewNoStackErrorf("the read_from_storage hint might not work"))
```

Reply:
And I think maybe "Optimizer Hint read_from_storage hint is inapplicable" is better since we already know this hint doesn't work, and this is also more consistent with other warnings for hints.